controller/metrics.py

```diff
@@ -1,56 +1,88 @@
+    sessions_status_changes: Counter = Counter(
+        "sessions_status_changes",
+        "Number of times a status change has occured",
+        labelnames=[
+            *config.METRICS_EXTRA_LABELS_SANITIZED,
+            "status_from",
+            "status_to",
+        ]
     )
-    num_of_running_sessions: Gauge = Gauge(
-        "sessions_running",
-        "Number of sessions in running state",
-        labelnames=config.METRICS_EXTRA_LABELS_SANITIZED,
+    sessions_launch_duration: Histogram = Histogram(
+        "sessions_launch_duration_seconds",
+        "How long did it take for a session to transition into running state",
+        unit="seconds",
+        labelnames=config.METRICS_EXTRA_LABELS_SANITIZED
     )
-    num_of_starting_sessions: Gauge = Gauge(
-        "sessions_starting",
-        "Number of sessions which have been scheduled but are still starting",
-        labelnames=config.METRICS_EXTRA_LABELS_SANITIZED,
+    sessions_cpu_request: Histogram = Histogram(
+        "sessions_cpu_request_millicores",
+        "CPU millicores requested by a user for a session.",
+        unit="m",
+        labelnames=config.METRICS_EXTRA_LABELS_SANITIZED
+    )
+    sessions_memory_request: Histogram = Histogram(
+        "sessions_memory_request_bytes",
+        "Memory requested by a user for a session.",
+        unit="byte",
+        labelnames=config.METRICS_EXTRA_LABELS_SANITIZED
+    )
+    sessions_disk_request: Histogram = Histogram(
+        "sessions_disk_request_bytes",
+        "Disk space requested by a user for a session.",
+        unit="byte",
+        labelnames=config.METRICS_EXTRA_LABELS_SANITIZED
     )
 
     def manipulate(
         self,
         metric_name: str,
         operation: str,
+        value: Union[int, float],
         manifest_labels: Dict[str, str] = {},
+        extra_labels: Dict[str, str] = {},
     ):
         metric = getattr(self, metric_name)
+        labels = {}
         if len(config.METRICS_EXTRA_LABELS) > 0:
-            labels = {}
             for i in range(len(config.METRICS_EXTRA_LABELS)):
                 try:
                     labels[config.METRICS_EXTRA_LABELS_SANITIZED[i]] = manifest_labels[
                         config.METRICS_EXTRA_LABELS[i]
                     ]
                 except KeyError:
                     # NOTE: We should not be super sensitive to k8s label changes, so if a
-                    # lable that is exepected to exist in the manifest does not then
+                    # label that is exepected to exist in the manifest does not then
                     # the metric label will not be applied, but the metric will be counted
                     labels[config.METRICS_EXTRA_LABELS_SANITIZED[i]] = "Unknown"
+        labels = {**labels, **extra_labels}
+        if len(labels.keys()) > 0:
             metric = metric.labels(**labels)
-        getattr(metric, operation)()
+        try:
+            getattr(metric, operation)(value)
```

Comment:
Should we first check `getattr(metric, operation, None) is not None`, with an error log entry like "Operation xyz doesn't exist" and only then do this try&except ?